packages/ejson/test/person.g.dart

```diff
@@ -0,0 +1,35 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
+
+part of 'person.dart';
+
+// **************************************************************************
+// EJsonGenerator
+// **************************************************************************
+
+EJsonValue encodePerson(Person value) {
+  return {
+    'name': value.name.toEJson(),
+    'birthDate': value.birthDate.toEJson(),
+    'income': value.income.toEJson(),
+    'spouse': value.spouse.toEJson()
+  };
+}
+
+Person decodePerson(EJsonValue ejson) {
+  return switch (ejson) {
+    {
+      'name': EJsonValue name,
+      'birthDate': EJsonValue birthDate,
+      'income': EJsonValue income,
+      'spouse': EJsonValue spouse
+    } =>
+      Person(fromEJson(name), fromEJson(birthDate), fromEJson(income),
+          spouse: fromEJson(spouse)),
+    _ => raiseInvalidEJson(ejson),
+  };
+}
```

Comment:
Should these be freestanding functions?

Reply:
Yes. The `ejson_generator` runs after `Person` is defined. We cannot _"re-open"_ a class and add more to it.

Reply:
Can we not add them in an extension on person?

Reply:
Obviously the decoding cannot be an extension method, but for encoding we do generate a `toEJson()` extension method:
```dart
extension PersonEJsonEncoderExtension on Person {
  @pragma('vm:prefer-inline')
  EJsonValue toEJson() => encodePerson(this);
}
```
The `encodePerson` is convenient when registering the codec pari:
```dart
register(encodePerson, decodePerson);
```
Otherwise you would have to write:
```dart
register((Person p) => p.toEJson(), decodePerson);
```

Reply:
Right - I guess we'd need https://github.com/dart-lang/language/issues/723 to be able to move the decode to an extension/factory. Should we generate a convenience registration method or is that too much - like:

```dart
void registerPersonEncoders() {
  register(encodePerson, decodePerson);
}
```

Reply:
There is [feature specification](https://github.com/dart-lang/language/blob/main/working/0107%20-%20implicit-constructors/feature-specification.md) for _implicit constructors_ from back in May that includes `static extension`s, but alas we don't have that yet.

Reply:
I like the generated _register_ function. Will go with:
```dart
void registerPerson() => register(_encodePerson, _decodePerson);
```
Note that the codec pair becomes private.

Reply:
Fixed